### DISCUSSION TITLE
Calculation of material property (permeability) as function of variable (temperature)

### DISCUSSION THREAD

**Discussion by RotmanC** [idaholab/moose#27488] [Q&A Modules: Porous Flow]
Hello,
I'm working on the evolution of supercritical conditions by a magmatic intrusion at depth (7km), so the permeability (_k_) changes with the evolution of this high temperature body (825K).

I implement an equation to calculate _k_ as function of temperature following the equation of Hayba & Ingebritsen (1997). I include it using the permeability as `PorousFlowPermeabilityTensorFromVar` that call an `AuxKernel` and its `AuxVariable`.

**The problem is that it doesn't converge**. May be because  the permeability has a big change (_k_ = [1e-22, 1e-14]), which is at the middle bottom of the mesh in a very short volume, that is, around the intrusion of magma (825K, _k_ = 1e-22). So the Jacobian has a wide range of values to deal.

I tried:
- Refine the mesh
- Smooth the BC of the intrusion of 825K
- Smooth the equation that computes the permeability
- Modify the Executer, Preconditions and time steps
- Include controls to start the BC of the intrusion late, after some timesteps
- Include `Controls` to start the calculation of _k_ after some timesteps after the starting of the intrusion
- Include and `Adaptivity`, but I cannot use _k_ as variable (and spend a lot of time running)

May be I can calculate _k_ as a MultiApp, or can I calculate outside and call the results later (but I don't know how to do it and I didn't found anything in the discussions).

Another question that I have is why the size of the initial execution of this simulation is more than 2Gb (2368 MB) if only runs 3570 elements in 2D.

Here is the [project ](https://gitfront.io/r/RotmanCM/4r5YRwDZbqx4/SC-evolution/) with the mesh and the script with no `Controls` and `Adaptivity`

Any support and comment to improve it are more than welcome

**Comment by RotmanC** [comment]
Actually I tried to use `execute_on = CUSTOM` too, but I don't understand how to include the list of times to compute _k_ (to avoid to create a custom execute flag). Is it posible?

**Comment by GiudGiud** [comment]
Hello

Does it converge if you set k to be a constant?
To troubleshoot convergence issues there are a list of techniques here:
https://mooseframework.inl.gov/moose/application_usage/failed_solves.html

Guillaume

**Reply by RotmanC** [reply]
Many thanks! **Now it works** :)

Also was helpful to check:
1. the Jacobian using [FiniteDifferencePreconditioner](https://mooseframework.inl.gov/moose/source/preconditioners/FiniteDifferencePreconditioner.html)
2. [the convergence criteria](https://mooseframework.inl.gov/modules/porous_flow/convergence.html)
3. [the numerical stabilization](https://mooseframework.inl.gov/modules/porous_flow/stabilization.html)

Also, I have some questions left about the use of `execute_on = CUSTOM`. Isn't it possible to include a list of time steps or include a range of time steps to avoid using to calculate a variable (for example, calculate it every 10 time steps) without creating a flag?

Many thanks for your help

**Reply by GiudGiud** [reply]
Hello

Execute_on = custom is not for selecting a list of time steps currently. 

What you can use for that is the Controls system

Guillaume
